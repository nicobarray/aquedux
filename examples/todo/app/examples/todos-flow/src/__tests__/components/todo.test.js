// @flow

import React from 'react';
import { shallow } from 'enzyme';

import Todo from '../../components/Todo';

const setup = (setupProps = {}) => {
  const defaultProps = {
    text: 'Test todo',
    completed: false,
    onClick: jest.fn()
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(
    <Todo
      text={props.text}
      completed={props.completed}
      onClick={props.onClick}
    />
  );

  return {
    props,
    wrapper
  };
};

describe('Todo', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  test('puts a line through text when completed', () => {
    const { wrapper } = setup({ completed: true });

    expect(wrapper).toMatchSnapshot();
  });

  test('calls onClick when clicked', () => {
    const { props, wrapper } = setup();
    expect(wrapper).toMatchSnapshot();

    wrapper.find('li').simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });
});
