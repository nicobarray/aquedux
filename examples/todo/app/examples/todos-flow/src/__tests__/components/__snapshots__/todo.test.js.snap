// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo calls onClick when clicked 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <li
    onClick={[Function]}
    style={
        Object {
            "textDecoration": "none",
          }
    }
>
    Test todo
</li>,
  "nodes": Array [
    <li
      onClick={[Function]}
      style={
            Object {
                  "textDecoration": "none",
                }
      }
>
      Test todo
</li>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Todo
        completed={false}
        onClick={[Function]}
        text="Test todo"
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "completed": false,
          "onClick": [Function],
          "text": "Test todo",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <li
          onClick={[Function]}
          style={
                    Object {
                              "textDecoration": "none",
                            }
          }
>
          Test todo
</li>,
        "_debugID": 6,
        "_renderedOutput": <li
          onClick={[Function]}
          style={
                    Object {
                              "textDecoration": "none",
                            }
          }
>
          Test todo
</li>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Todo
    completed={false}
    onClick={[Function]}
    text="Test todo"
/>,
}
`;

exports[`Todo puts a line through text when completed 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <li
    onClick={[Function]}
    style={
        Object {
            "textDecoration": "line-through",
          }
    }
>
    Test todo
</li>,
  "nodes": Array [
    <li
      onClick={[Function]}
      style={
            Object {
                  "textDecoration": "line-through",
                }
      }
>
      Test todo
</li>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Todo
        completed={true}
        onClick={[Function]}
        text="Test todo"
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "completed": true,
          "onClick": [Function],
          "text": "Test todo",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <li
          onClick={[Function]}
          style={
                    Object {
                              "textDecoration": "line-through",
                            }
          }
>
          Test todo
</li>,
        "_debugID": 4,
        "_renderedOutput": <li
          onClick={[Function]}
          style={
                    Object {
                              "textDecoration": "line-through",
                            }
          }
>
          Test todo
</li>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Todo
    completed={true}
    onClick={[Function]}
    text="Test todo"
/>,
}
`;

exports[`Todo renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <li
    onClick={[Function]}
    style={
        Object {
            "textDecoration": "none",
          }
    }
>
    Test todo
</li>,
  "nodes": Array [
    <li
      onClick={[Function]}
      style={
            Object {
                  "textDecoration": "none",
                }
      }
>
      Test todo
</li>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Todo
        completed={false}
        onClick={[Function]}
        text="Test todo"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "completed": false,
          "onClick": [Function],
          "text": "Test todo",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <li
          onClick={[Function]}
          style={
                    Object {
                              "textDecoration": "none",
                            }
          }
>
          Test todo
</li>,
        "_debugID": 2,
        "_renderedOutput": <li
          onClick={[Function]}
          style={
                    Object {
                              "textDecoration": "none",
                            }
          }
>
          Test todo
</li>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Todo
    completed={false}
    onClick={[Function]}
    text="Test todo"
/>,
}
`;
